src/unix/fsevents.c

```diff
@@ -267,12 +271,6 @@ static void uv__fsevents_event_cb(ConstFSEventStreamRef streamRef,
         }
       }
 
-#ifdef MAC_OS_X_VERSION_10_7
-      /* Ignore events with path equal to directory itself */
-      if (len == 0)
```

Comment:
ok, I've been messing with this a bit and I cannot reproduce the extra event for directories.

I am using the following https://gist.github.com/evanlucas/8db596602f2bb693b541.

run `mkdir -p watch_dir && ./watchdir`

and then in another shell, `touch watch_dir/file1`

It will only emit a single event with the path being `file1` (on OS X 10.11 at least). I haven't tested on anything older